core/src/main/java/com/google/instrumentation/trace/SpanData.java

```diff
@@ -0,0 +1,278 @@
+      TimedEvents<Annotation> annotations,
+      TimedEvents<NetworkEvent> networkEvents,
+      Links links,
+      @Nullable Status status,
+      @Nullable Timestamp endTimestamp) {
+    return new AutoValue_SpanData(
+        checkNotNull(context, "context"),
+        parentSpanId,
+        checkNotNull(displayName, "displayName"),
+        checkNotNull(startTimestamp, "startTimestamp"),
+        checkNotNull(attributes, "attributes"),
+        checkNotNull(annotations, "annotations"),
+        checkNotNull(networkEvents, "networkEvents"),
+        checkNotNull(links, "links"),
+        status,
+        endTimestamp);
+  }
+
+  /**
+   * Returns the {@code SpanContext} associated with this {@code Span}.
+   *
+   * @return the {@code SpanContext} associated with this {@code Span}.
+   */
+  public abstract SpanContext getContext();
+
+  /**
+   * Returns the parent {@code SpanId} or {@code null} if the {@code Span} is a root {@code Span}.
+   *
+   * @return the parent {@code SpanId} or {@code null} if the {@code Span} is a root {@code Span}.
+   */
+  @Nullable
+  public abstract SpanId getParentSpanId();
+
+  /**
+   * Returns the display name of this {@code Span}.
+   *
+   * @return the display name of this {@code Span}.
+   */
+  public abstract String getDisplayName();
+
+  /**
+   * Returns the start {@code Timestamp} of this {@code Span}.
+   *
+   * @return the start {@code Timestamp} of this {@code Span}.
+   */
+  public abstract Timestamp getStartTimestamp();
+
+  /**
+   * Returns the attributes recorded for this {@code Span}.
+   *
+   * @return the attributes recorded for this {@code Span}.
+   */
+  public abstract Attributes getAttributes();
+
+  /**
+   * Returns the annotations recorded for this {@code Span}.
+   *
+   * @return the annotations recorded for this {@code Span}.
+   */
+  public abstract TimedEvents<Annotation> getAnnotations();
+
+  /**
+   * Returns network events recorded for this {@code Span}.
+   *
+   * @return network events recorded for this {@code Span}.
+   */
+  public abstract TimedEvents<NetworkEvent> getNetworkEvents();
+
+  /**
+   * Returns links recorded for this {@code Span}.
+   *
+   * @return links recorded for this {@code Span}.
+   */
+  public abstract Links getLinks();
+
+  /**
+   * Returns the {@code Status} or {@code null} if {@code Span} is still active.
+   *
+   * @return the {@code Status} or {@code null} if {@code Span} is still active.
+   */
+  @Nullable
+  public abstract Status getStatus();
+
+  /**
+   * Returns the end {@code Timestamp} or {@code null} if the {@code Span} is still active.
+   *
+   * @return the end {@code Timestamp} or {@code null} if the {@code Span} is still active.
+   */
+  @Nullable
+  public abstract Timestamp getEndTimestamp();
+
+  SpanData() {}
+
+  /** A timed event representation. It can be a timed {@link Annotation} or {@link NetworkEvent}. */
+  @Immutable
+  @AutoValue
+  public abstract static class TimedEvent<T> {
```

Comment:
add package private constructor noting that types of TimedEvent are sealed, ex only defined here. This will make it easier to parse/marshal.